README.md

```diff
@@ -0,0 +1,81 @@
+# ❄️+🪁 Generate Buildkite Pipelines from Nix Flakes
+
+This repository contains some functions that'll transform a Nix flake into a
+valid BuildKite Pipeline, represented in JSON.
+
+## 🧑‍💻 Usage
+
+We'll take a look at a minimal example flake that “builds” a packages named
+`hello`, and “checks” that the output of running the `hello` package matches
+`Hello, world!`. Here is that flake.
+
+```nix
+{
+  description = "flake-buildkite-pipeline example";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-22.05";
+    flake-buildkite-pipeline.url = "github:tweag/flake-buildkite-pipeline";
+  };
+
+  outputs = { self, nixpkgs, flake-buildkite-pipeline }:
+    let
+      system = "x86_64-linux";
+      pkgs = nixpkgs.legacyPackages.${system};
+    in {
+      packages.${system}.hello = pkgs.hello;
+
+      checks.${system}.demoCheck = pkgs.runCommandNoCC "demoCheck" {
+        meta.checkDescription = "demoCheck";
+      } ''
+        set -o verbose
+        want='Hello, world!'
+        got="$(${self.packages.${system}.hello}/bin/hello)"
+        [ "$want" == "$got" ]
+        touch $out
+      '';
+
+      pipelines.buildkite.steps = flake-buildkite-pipeline.lib.flakeSteps {
+        commonExtraStepConfig = {
+          agents = [ "nix" ];
+          plugins = [{ "thedyrt/skip-checkout#v0.1.1" = null; }];
+        };
+      } self;
+    };
+}
+```
+
+Apart from the `pipelines` attribute, everything looks like a normal flake, and
+we can still use it a such. Try running `nix flake check`, and you can observe
+that it builds the `hello` package, and runs the singular check.
+
+Now, to get the actual BuildKite `pipeline.json`, run `nix eval --json
+.#pipelines.buildkite`. This will produce a valid BuildKite pipeline, the
+`--json` flag will print it out as JSON instead of Nix. You can use this in
+combination with the BuildKite Agent CLI tool to upload the pipeline, e.g.
+
+```shell
+nix eval "git+file://$(pwd)?submodules=1"#pipeline.buildkite --json | buildkite-agent pipeline upload
```

Comment:
The submodule stuff is not needed in general. Also, you named the output `pipelines.buildkite` . A better command would be

```suggestion
nix eval .#pipelines.buildkite --json | buildkite-agent pipeline upload
```

Reply:
Makes sense, could we perhaps even do something along the lines of this? Once it merged into master.

```suggestion
nix eval "github:tweag/flake-buildkite-pipeline?dir=examples"#pipelines.buildkite --json | buildkite-agent pipeline upload
```

Reply:
I think it's a bit confusing having a direct reference to this repo in the documentation. If I was just skimming through it, I would assume that I can just copy-paste this into my CI and have it build _my_ flake, not the example one we provide here.

Reply:
Okay, that way yours does indeed make more sense. I'll commit that one, thanks. 😊